src/lib/components/EnhancedSelect/index.js

```diff
@@ -0,0 +1,476 @@
+   * sets the value to the entries.
+   * @param {*} state
+   */
+  onStoreUpdate(state) {
+    const filters = state.filters || {};
+    const entries = filters[this.name] || [];
+    this.setValue(entries.map((entry) => entry.value));
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    this.removeEventListener('focusout', this.handleFocusOut);
+
+    this._elements.label.removeEventListener('click', this.handleLabelClick);
+    this._elements.label.removeEventListener(
+      'keydown',
+      this.handleLabelKeydown,
+    );
+    Array.from(this._elements.listItems).forEach((item) =>
+      item.removeEventListener('click', this.handleSelection),
+    );
+    this._elements.list.removeEventListener('keydown', this.handleSelection);
+  }
+
+  /**
+   * @param {String[]} value
+   */
+  setValue(value) {
+    this.value = value;
+
+    this.options.forEach((o) => (o.selected = value.includes(o.value)));
+
+    const data = new FormData();
+
+    value.forEach((value) => data.append(this.name, value));
+
+    // ElementInternals are not yet supported in Safari, but would
+    // also only be used in a <form> context, where we currently
+    // don't use it.
+    if (this.internals) {
+      this.internals.setFormValue(data);
+    }
+
+    this.dispatchEvent(new Event('change', {bubbles: true, cancelable: true}));
+  }
+
+  render() {
+    return html`
+      <div
+        class="enhanced-select__wrapper display-flex align-center"
+        ?open="${this.open}"
+      >
+        <label
+          id="${this._labelId}"
+          class="display-flex align-center justify-content-between"
+          aria-controls="${this._dropdownId}"
+          tabindex="0"
+          @click="${this.handleLabelClick}"
+          @keydown="${this.handleLabelKeydown}"
+        >
+          <span>${this.displayLabel}</span>
+          <svg
+            width="10"
+            height="5"
+            viewBox="0 0 10 5"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path d="M0 0L5 5L10 0L0 0Z" fill="currentColor" />
+          </svg>
+        </label>
+
+        <ul
+          id="${this._dropdownId}"
+          class="enhanced-select__options"
+          @keydown="${this.handleListKeydown}"
+        >
+          ${this.options.map((option) => {
+            const selected = this.value?.includes(option.value);
+
+            return html`
+              <li
+                class="width-full gap-bottom-100 display-flex align-center justify-content-between"
+                id="${option.id}"
+                tabindex="0"
+                @click="${(event) => {
+                  this.handleSelection(event);
+                }}"
+                ?selected="${selected}"
+              >
+                <input
+                  type="checkbox"
+                  .value="${option.value}"
+                  .checked="${selected}"
+                />
```

Comment:
Not too fond of this, but thought it was the simpler way to get to match the style without changing the components too much from DCC. 

Happy to change it, though.